testing/src/test/java/io/stargate/it/CQLTest.java

```diff
@@ -187,59 +189,126 @@ public void querySystemLocal() throws UnknownHostException {
     assertThat(rows.next().getInetAddress("listen_address")).isIn(getStargateInetSocketAddresses());
   }
 
+  private Statement<?> insertIntoStatement(boolean prepared, boolean bindByName, String... args) {
+    assert args.length > 0 : "Expecting at least the key";
+    assert args.length <= 3 : "Expecting at most the key, value1 and value2";
+
+    if (prepared) {
+      PreparedStatement insertPrepared = session.prepare(insertIntoQuery());
+      BoundStatement statement;
+      if (bindByName) {
+        statement = insertPrepared.bind().setString("key", args[0]);
+        for (int i = 1; i < args.length; i++) {
+          statement = statement.setString("value" + i, args[i]);
+        }
+      } else {
+        statement = insertPrepared.bind((Object[]) args);
+      }
+      return statement;
+    } else {
+      SimpleStatementBuilder insertBuilder = SimpleStatement.builder(insertIntoQuery());
+      if (bindByName) {
+        insertBuilder.addNamedValue("key", args[0]);
+        for (int i = 1; i < args.length; i++) {
+          insertBuilder.addNamedValue("value" + i, args[i]);
+        }
+      } else {
+        insertBuilder.addPositionalValues((Object[]) args);
+      }
+      return insertBuilder.build();
+    }
+  }
+
+  private Statement<?> selectByKeyStatement(boolean prepared, boolean bindByName, String key) {
+    if (prepared) {
+      PreparedStatement selectPrepared = session.prepare(selectFromQuery(true));
+      BoundStatement statement;
+      if (bindByName) {
+        statement = selectPrepared.bind().setString("key", key);
+      } else {
+        statement = selectPrepared.bind(key);
+      }
+      return statement;
+    } else {
+      SimpleStatementBuilder selectBuilder = SimpleStatement.builder(selectFromQuery(true));
+      if (bindByName) {
+        selectBuilder.addNamedValue("key", key);
+      } else {
+        selectBuilder.addPositionalValue(key);
+      }
+      return selectBuilder.build();
+    }
+  }
+
   @Test
   public void querySimple() {
-    querySimple(false);
+    querySimple(false, false);
   }
 
   @Test
-  public void querySimpleBidingMarkersByName() {
-    querySimple(true);
+  public void querySimpleBindingByName() {
+    querySimple(false, true);
   }
 
-  private void querySimple(boolean bindMarkersByName) {
+  private void querySimple(boolean usePrepared, boolean bindByName) {
```

Comment:
nit: with JUnit 5 we could convert this method to a `@ParameterizedTest` with `@CsvSource` for example.

Reply:
also `unsetValues`

Reply:
Good idea. I ended up using an enum and `@EnumSource`. I think that's a tad more readable here.